packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -109,6 +115,15 @@ class CupertinoNavigationBar extends StatelessWidget implements ObstructingPrefe
   /// behind it.
   final Color backgroundColor;
 
+  /// The bottom border color of the navigation bar.
+  ///
+  /// If a border color is null or transparent, the tab bar will not display a border.
+  final Color borderColor;
+
+  /// Adjust system UI overlay style to [SystemUiOverlayStyle.light] or
+  /// [SystemUiOverlayStyle.dark] dynamically based off a background's luminance.
```

Comment:
Can you explain this in more detail?

Reply:
maybe give an example of when you would set this to false

Reply:
On screenshot above you can see dark and bright colors. Imagine 2 screens using those colors: 1st screen is dark color, 2nd screen is bright.

My iOS app is set by default (in XCode project) to use light theme for status bar. When app is launched and 1st screen with dark color is shown, I see SystemUIOverlayStyle.light. Once I navigate to 2nd screen with bright color, navigation bar recalculates background color and sets SystemUIOverlayStyle.dark. Once navigated back to 1st screen with dark color, I still see SystemUIOverlayStyle.dark.

It's most likely is bug, but I do believe as a customer I should be able to disable auto-calculation of background luminance and enforce particular style of status bar.

Hope this helps.

Reply:
If the only reason for doing this is that we're currently doing the wrong thing, we should just fix it to do the right thing, rather than adding a feature.

There's a cost to every feature -- documentation, tutorials, docs, compilation time, download time, etc.

Reply:
@Hixie what's the right thing to do here? Should I remove auto detection of luminance completely?

Reply:
Bring this up again after pushing addressed changes.

Reply:
The right thing to do is to fix https://github.com/flutter/flutter/issues/4630

Reply:
Would it be acceptable to keep this feature in the PR until the original issue is resolved? Or would you like me to remove the feature from this PR?

Thanks.